src/components/ItemList/ItemList.tsx

```diff
@@ -1,73 +1,76 @@
-import { useState, useEffect, useCallback, createContext } from 'react';
-import { NavLink, Outlet, useNavigate, useParams } from 'react-router-dom';
+import { useState, useEffect, createContext, useContext } from 'react';
+import { Outlet, useNavigate, useParams } from 'react-router-dom';
 import classes from './ItemList.module.css';
 import Paginator from '../Paginator/Paginator';
 import Loader from '../Loader/Loader';
 import { queryItems } from '../API/API';
-import { EMPTY_ITEMS_ARRAY } from './data';
-import { SEARCH_DEFAULT } from '../../data';
-import { MAX_AMOUNT } from '../../data';
+import {
+  SEARCH_DEFAULT,
+  MAX_AMOUNT,
+  DEFAULT_PAGE_NUMBER,
+  DEFAULT_ITEMS_PER_PAGE,
+  IGeneralContext,
+} from '../../shared/data/data';
+import Item from '../Item/Item';
+import Matches from '../Matches/Matches';
+import { GeneralContext } from '../MainLayout/MainLayout';
 
 export const DataFromChildContext = createContext<React.Dispatch<
   React.SetStateAction<boolean>
 > | null>(null);
 
 const ItemList = function () {
-  const [beerList, setBeerList] = useState(EMPTY_ITEMS_ARRAY);
   const [isLoading, setIsLoading] = useState(false);
   const [requestOK, setRequestOK] = useState(true);
   const [sectionOpen, setSectionOpen] = useState(false);
-  const [searchText, setSearchText] = useState('');
-  const [pageNumber, setPageNumber] = useState(1);
-  const [perPage, setPerPage] = useState(4);
+  const [pageNumber, setPageNumber] = useState(DEFAULT_PAGE_NUMBER);
+  const [perPage, setPerPage] = useState(DEFAULT_ITEMS_PER_PAGE);
 
   const navigate = useNavigate();
 
   const {
-    pageNum = 1,
+    pageNum = DEFAULT_PAGE_NUMBER,
     searchStr = localStorage.getItem(SEARCH_DEFAULT) || '',
     index,
   } = useParams();
 
-  const queryData = useCallback(
-    async (pageNumber: number) => {
+  const genContext: IGeneralContext | null = useContext(GeneralContext);
+
+  useEffect(() => {
+    if (searchStr) {
+      genContext?.setMainString(searchStr);
+    } else {
+      genContext?.setMainString('');

```

Comment:
You do the same on lines 41 and 43
you can do
`const searchStringToSave = searchStr || ""`
`genContext?.setMainString(searchStringToSave );`